### PR TITLE
feat: enable seeding of operator information and test data

### PR DESCRIPTION
**Pull request by Phil91**
### Set initial data

* make operator company BPN, name, shortname and address overwriteable

### Test data

* defined a default test data set within configMap and enabled its seeding in a configurable manner 
* add configuration to add own config maps for test data seeding

## Why

To be able to add custom test data and have the possibility to set the bpn for the operator via the configuration

## Issue

Refs: #449 #447

## Corresponding Backend PR

https://github.com/eclipse-tractusx/portal-backend/pull/1067

## Checklist

- [x] I have performed a self-review of my changes
- [x] I have successfully tested my changes
- [x] I have added comments in the default values.yaml file with helm-docs syntax ('# -- ') if relevant for installation
- [x] I have commented my changes, particularly in hard-to-understand areas

### CODE REVIEW

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+*/}}
+
+{{- if .Values.backend.portalmigrations.seeding.useInitial -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name:  "{{ include "portal.fullname" . }}-default-initialdata"
+  namespace: {{ .Release.Namespace }}
+data:
+  addresses.test.json: |-
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:27]: I think .test. isn't supposed to be used

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        "city": "Munich",
+        "region": null,
+        "streetadditional": null,
+        "streetname": "Street",
+        "streetnumber": "1",
+        "zipcode": "00001",
+        "country_alpha2code": "DE"
+      }
+    ]
+  companies.test.json: |-
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:42]: I think .test. isn't supposed to be used

File: charts/portal/templates/configmap-backend-seeding-testdata.yaml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+*/}}
+
+{{- if .Values.backend.portalmigrations.seeding.seedTestData.useDefault -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name:  "{{ include "portal.fullname" . }}-default-testdata"
+  namespace: {{ .Release.Namespace }}
+data:
+  addresses.test.json: |-
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-testdata.yaml:27]: I think .test. isn't supposed to be used - valide for all table in this file

File: charts/portal/templates/job-backend-portal-migrations.yaml [hunk 4]
```diff
@@ -70,33 +70,76 @@ spec:
...
+            name: "{{ include "portal.fullname" . }}-default-testdata"
+            optional: true
+        {{- end }}
+        {{- if and (.Values.backend.portalmigrations.seeding.seedTestData.useOwnConfigMap.configMap) (.Values.backend.portalmigrations.seeding.seedTestData.useOwnConfigMap.filename) }}
+        - name: test-data-custom
+          configMap:
+            name: "{{ .Values.backend.portalmigrations.seeding.seedTestData.useOwnConfigMap.configMap }}"
+            optional: true
+        {{- end }}
+      {{- end }}
```
Comment (evegufy) [inline charts/portal/templates/job-backend-portal-migrations.yaml:145]: please add empty line

File: charts/portal/templates/job-backend-portal-migrations.yaml [hunk 5]
```diff
@@ -70,33 +70,76 @@ spec:
...
-            value: "{{ .Values.backend.portalmigrations.seeding.testDataEnvironments }}"
           - name: "SEEDING__DATAPATHS__0"
-            value: "Seeder/Data"
+            value: "Seeder/Data/"
+          - name: "SEEDING__TESTDATAENVIRONMENTS__0"
+            value: ""
+          - name: "SEEDING__DATAPATHS__1"
+            value: "Seeder/Data/overwrite"
+          - name: "SEEDING__TESTDATAENVIRONMENTS__1"
+            value: ""
```
Comment (evegufy) [inline charts/portal/templates/job-backend-portal-migrations.yaml:80]: empty value?

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        "id": "b4db3945-19a7-4a50-97d6-e66e8dfd04fb",
+        "date_created": "2022-03-24 18:01:33.306000 +00:00",
+        "date_last_changed": "2022-03-24 18:01:33.306000 +00:00",
+        "city": "Munich",
+        "region": null,
+        "streetadditional": null,
+        "streetname": "Street",
+        "streetnumber": "1",
+        "zipcode": "00001",
+        "country_alpha2code": "DE"
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:39]: values should be enabled for dynamic seeding

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    ]
+  companies.test.json: |-
+    [
+      {
+        "id": "2dc4249f-b5ca-4d42-bef1-7a7a950a4f87",
+        "date_created": "2022-03-24 18:01:33.306000 +00:00",
+        "business_partner_number": "{{ .Values.operator.bpn }}",
+        "name": "Catena-X",
+        "shortname": "Catena-X",
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:49]: values should be enabled for dynamic seeding

File: charts/portal/values.yaml [hunk 8]
```diff
@@ -618,13 +622,17 @@ backend:
...
+      # -- If set to true the data configured in the config map 'configmap-backend-seeding-initialdata.yaml' will be taken to insert the initial data
+      useInitial: false
       # -- Option to seed test data provided in a configMap
-      testData:
-        # -- ConfigMap containing json files for the tables to seed, e.g. companies.test.json, addresses.test.json, etc.
-        configMap: ""
-        # -- Filename identifying the test data files e.g. for companies.test.json the value would be "test"
-        filename: ""
+      seedTestData:
+        # -- If set to true the data configured in the config map 'configmap-backend-seeding-testdata.yaml' will be taken to insert the default test data
```
Comment (evegufy) [inline charts/portal/values.yaml:629]: Please update comment to [here](https://github.com/SAP/ssi-dim-middle-layer/blob/main/charts/dim/values.yaml#L66)

**Review by evegufy — CHANGES_REQUESTED** [review]

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 1 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: fixed

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 2 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: fixed

File: charts/portal/templates/configmap-backend-seeding-testdata.yaml (hunk 3 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-testdata.yaml]: fixed

File: charts/portal/templates/job-backend-portal-migrations.yaml (hunk 4 above)
Comment (Phil91) [reply charts/portal/templates/job-backend-portal-migrations.yaml]: added

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    ]
+  companies.json: |-
+    [
+      {
+        "id": "2dc4249f-b5ca-4d42-bef1-7a7a950a4f87",
+        "date_created": "2022-03-24 18:01:33.306000 +00:00",
+        "business_partner_number": "{{ .Values.operator.bpn }}",
+        "name": "Catena-X",
+        "shortname": "Catena-X",
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:49]: Default values:
```suggestion
        "name": "CX-Operator",
        "shortname": "CX-Operator",
```

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        "id": "b4db3945-19a7-4a50-97d6-e66e8dfd04fb",
+        "date_created": "2022-03-24 18:01:33.306000 +00:00",
+        "date_last_changed": "2022-03-24 18:01:33.306000 +00:00",
+        "city": "Munich",
+        "region": null,
+        "streetadditional": null,
+        "streetname": "Street",
+        "streetnumber": "1",
+        "zipcode": "00001",
+        "country_alpha2code": "DE"
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:39]: Default values:
```suggestion
        "city": "tdb",
        "region": "",
        "streetadditional": "",
        "streetname": "tbd",
        "streetnumber": "",
        "zipcode": "",
        "country_alpha2code": "DE"
```

**Review by evegufy — CHANGES_REQUESTED** [review]

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 9 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: adjusted

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 10 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: adjusted

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 6 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: adjusted

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 7 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: adjusted

File: charts/portal/values.yaml [hunk 11]
```diff
@@ -618,13 +622,18 @@ backend:
         cpu: 75m
         memory: 350M
     seeding:
-      testDataEnvironments: ""
+      # -- Enables dynamic seeding of bpn (dim.bpn) and did document location address (dim.didDocumentLocationAddress) of the operator;
```
Comment (evegufy) [inline charts/portal/values.yaml:625]:
```suggestion
      # -- Enables dynamic seeding of information related to the operator company: operator.bpn, operator.name, operator.city etc.;
```

File: charts/portal/values.yaml [hunk 12]
```diff
@@ -618,13 +622,18 @@ backend:
...
+      # If set to `true` the data configured in the config map 'configmap-seeding-initialdata.yaml' will be taken to insert the initial data;
+      useInitial: false
```
Comment (evegufy) [inline charts/portal/values.yaml:627]: should default to true
```suggestion
      useInitial: true
```

File: charts/portal/values.yaml [hunk 13]
```diff
@@ -85,6 +85,10 @@ dimWrapper:
   apiPath: "/api/dim"
 decentralIdentityManagementAuthAddress: "https://dis-integration-service-prod.eu10.dim.cloud.sap/api/v2.0.0/iatp/catena-x-portal"
 
+operator:
```
Comment (evegufy) [inline charts/portal/values.yaml:88]:
```suggestion

# -- Set information related to the operator company
operator:
```

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+      {
+        "id": "b4db3945-19a7-4a50-97d6-e66e8dfd04fb",
+        "date_created": "2022-03-24 18:01:33.306000 +00:00",
+        "date_last_changed": "2022-03-24 18:01:33.306000 +00:00",
+        "city": "tbd",
+        "region": "",
+        "streetadditional": "",
+        "streetname": "tbd",
+        "streetnumber": "",
+        "zipcode": "",
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:38]: Please enable those values in the values.yaml file under operator.

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    ]
+  companies.json: |-
+    [
+      {
+        "id": "2dc4249f-b5ca-4d42-bef1-7a7a950a4f87",
+        "date_created": "2022-03-24 18:01:33.306000 +00:00",
+        "business_partner_number": "{{ .Values.operator.bpn }}",
+        "name": "CX-Operator",
+        "shortname": "CX-Operator",
```
Comment (evegufy) [inline charts/portal/templates/configmap-backend-seeding-initialdata.yaml:49]: Please enable those values in the values.yaml file under operator.

**Review by evegufy — CHANGES_REQUESTED** [review]

File: charts/portal/values.yaml (hunk 12 above)
Comment (Phil91) [reply charts/portal/values.yaml]: adjusted

File: charts/portal/values.yaml (hunk 13 above)
Comment (Phil91) [reply charts/portal/values.yaml]: adjusted

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 14 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: done

File: charts/portal/templates/configmap-backend-seeding-initialdata.yaml (hunk 15 above)
Comment (Phil91) [reply charts/portal/templates/configmap-backend-seeding-initialdata.yaml]: done

**Pull request merged**